### DISCUSSION TITLE
Flows Notifications - add collection/item link functionality

### DISCUSSION THREAD

**Discussion by w0kyj** [directus/directus#18705] [Feature Requests]
### Summary

Update the Flows Notification operation with ability to configure the linked collection and item. Currently notifications from a Flow do not have a link to a record and when clicked don't do anything, unlike a comment notification that takes the user to the item.

### Basic Example

Add parameter(s) to the notification operation for collection & item:
directus_notifications:
- Comment record example:
```json
{
"id": 67,
"timestamp": "2022-08-25T15:19:51.025Z",
"status": "inbox",
"recipient": "ba4b46bc-6147-47d0-9e66-cbe93328274c",
"sender": "ba4b46bc-6147-47d0-9e66-cbe93328274c",
"subject": "You were mentioned in articles",
"message": "\nHello Admin User,\n\nAdmin User has mentioned you in a comment:\n\n>  <em>Admin User</em> please review\n\n<a href=\"https://sales-demo-1.directus.app/admin/content/articles/3\">Click here to view.</a>\n",
"collection": "articles",
"item": "3"
}
```

Current Flow Notification example - missing collection & item:
```json
{
"id": 134,
"timestamp": "2022-10-13T13:09:06.163Z",
"status": "archived",
"recipient": "ba4b46bc-6147-47d0-9e66-cbe93328274c",
"sender": "ba4b46bc-6147-47d0-9e66-cbe93328274c",
"subject": "New Article Ready for Review",
"message": "Article is ready to review.",
"collection": null,
"item": null
}
```

### Motivation

The current notifications for Studio App users doesn't provide any details or link to the item that requires their attention. There is a workaround using an update operation after the notification to add the collection and item, but this is not ideal.

By adding the collection and item values as part of the operation, it is clear to the administrators setting up the flows and makes the notification usable for the content editors.

### Detailed Design

Update the Flow Notification operation API and Studio App UX to include two additional fields: collection and item where one collection and one item key may be set:
collection = collections.collection
item = <collection>.<item_pk>

The collection and item values may be set using any valid Flow variables available at this flow point:
Collection(string) = {{$trigger.body.collection}}
Item(string) = {{$trigger.body.keys[0]}}

### Must Have:
- Configuration options for linking the notification to the desired Collection & Item
- The collection and item values may be set using any valid Flow variables(collection name and item pk) available at this flow point
- The new configuration options must continue to allow null values for backward compatibility of existing notification operations.

### Should Have:
- pre-fill template variables for the default trigger collection and first item
    - Collection(string) = {{$trigger.body.collection}}
    - Item(string) = {{$trigger.body.keys[0]}}

### Could Have:
- Configuration option to choose an email template if custom email templates available
    - Make this a pick-list or searchable list of available custom templates
    - Default Value = Default System Template

### Won't Have:
- Changes to notifications inbox (eg show body message, mark read, ...)
- Changes to comments (eg configurable subject, ...)

### Drawbacks

Only real drawback is the cost and time to implement.

### Alternatives

https://github.com/directus/directus/issues/13727#issuecomment-1401788353
sure, I have created an additional step ("Daten aktualisieren") to my flow after sending a push notification ("Push senden").
in the step to send a push notification the name of the key is important. this is used as reference in the step to update the data. in my case this is notification_c66o5 (you can rename it if you want). to update a push notification I needed to configure a "updater" with the collection "directus_notifications" and a specific id. this id is contained in the payload of the previous step under the mentioned key notification_c66o5, which is passed as an array. so the specific id our push notification can be retrieved per {{notification_c66o5[0]}}. now the data need to be passed to the collection in order to update it. in my case I set the collection and item from the values in the $trigger variable.

{
  "collection": "{{ $trigger.collection }}",
  "item": "{{ $trigger.key }}"
}
this way push notifications with collection: null and item: null get necessary values, that will be used by navigating when clicking in the overview of notifications.
![image](https://github.com/directus/directus/assets/40695629/3cde63fb-0990-4cf7-abf4-b88070d9136b)

### Adoption Strategy

- Administrators may simply update any existing Flows with notifications to add the correct collection and item values
- No migrations or breaking changes expected

### Unresolved Questions

n/a

**Comment by w0kyj** [comment]
Possible related issue with bulk export notifications:
https://github.com/directus/directus/issues/18841

**Comment by abernh** [comment]
Yes, second that.
I'm using the notifications to inform staff members with a formatted email about new form entries. The email itself has a link to the new item but the notification item in the in-app-notifications are missing that important information as the notifications contents are not accessible here.

> There is a workaround using an update operation after the notification to add the collection and item, but this is not ideal.

Thanks for that mentioned workaround. Implementing it right away.
